### PR TITLE
Collected food fixes

### LINKED ISSUE #78167: Liberator Golems can't eat
## Reproduction: Spawned in as the free golems that worship "The Liberator", tried to eat raw iron, regular iron, mushrooms, different minerals. Could not eat, and eventually statued and died from starvation. I tried targeting the mouth, head, and chest when I ate, as well as feeding the other golem that spawned along with me. The other golem was unable to eat nor feed me either.
I don't know if it's a problem with all golems, or just the deathspawn ones.

An admin did feed me once to revive, and I tried spawning in a second time as another golem to try again, but eventually starved the same way.
sadgolems

### LINKED ISSUE #78192: Cooked food is completely fucking broken
Reporting client version: 514.1589

## Round ID:
[214161](https://scrubby.melonmesa.com/round/214161)

## Testmerges:
- [Revert "Bumps rust-g to `3.0.0` "](https://www.github.com/tgstation/tgstation/pull/77307)

## Reproduction:

- Cook food
- Expect to be able to interact with the food
- Be completely unable to pick up or do anything with it because it's not getting the init flag for some reason??????

image

Even when I var edit the init flag on correctly, I still can't actually eat it because its reagents list is also nulled, so something _really_ fucky is going on here.

### LINKED ISSUE #78194: Lots of food items cant be clicked by observers
## Reproduction:

Step 1: Observe
Step 2: Cook a donk pocket
Step 3: Try examining clicking or orbitting: Nothing happens

This is because of this runtime:

image

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

Fixes #78167
Fixes #78192
Fixes #78194

I went through the code and tried to find all of the remaining places we forgot to update the arguments passed into `item/food/Initialize` after more arguments were added to it, because there were a couple and they caused things to stop working.
Most notably, golems were unable to eat anything because nothing would correctly spawn "golem food".

_Additionally_ we were using a bunch of named arguments in new whenever crafting or cooking food. This runtimed, causing the food not to init properly.
_On top of that_ a late code review on a recent PR processed a list into a string_assoc_list twice causing it to become null.

Finally, we were trying to check the food preferences of examining ghosts or dogs or other non-human mobs. We shouldn't do that.

I also added a unit test for moth and golem food in the hopes that we'll notice them breaking.

## Changelog

:cl:
fix: Golems can eat
fix: Cooked and crafted food should be edible
fix: Medborgs can now spawn vanilla ice cream instead of nothing ice cream
fix: Ghosts can examine food
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
image
i'm not owned i'm not owned

File: code/game/objects/items/robot/items/food.dm [hunk 1]
```diff
@@ -60,7 +60,12 @@
 		if(DISPENSE_LOLLIPOP_MODE)
 			food_item = new /obj/item/food/lollipop/cyborg(turf_to_dispense_to)
 		if(DISPENSE_ICECREAM_MODE)
-			food_item = new /obj/item/food/icecream(turf_to_dispense_to, list(ICE_CREAM_VANILLA))
+			food_item = new /obj/item/food/icecream(
+				/* loc = */ turf_to_dispense_to,
+				/* starting_reagent_purity = */ null,
+				/* no_base_reagents = */ FALSE,
+				/* prefill_flavours = */ list(ICE_CREAM_VANILLA),
```
Comment (ZephyrTFA) [inline code/game/objects/items/robot/items/food.dm:67]: get rid of these comment markers, just used actual named arguments

Comment (Jacquerel) [reply code/game/objects/items/robot/items/food.dm:67]: These named arguments we _already have_ in the codebase are runtiming (so I need to make them not named in this PR). I swear to you this shit doesn't work.
image

Comment (ZephyrTFA) [reply code/game/objects/items/robot/items/food.dm:67]: discussed w/ spaceman

**Comment by carshalash** [comment]
There seems to be an issue with popsicle sticks at the moment. Upon creation you just cant interact with them with left click.

**Comment by Jacquerel** [comment]
> There seems to be an issue with popsicle sticks at the moment. Upon creation you just cant interact with them with left click.

this should fix that

**Comment by carshalash** [comment]
> > There seems to be an issue with popsicle sticks at the moment. Upon creation you just cant interact with them with left click.
> 
> this should fix that

Tested on local, all good.

**Pull request merged**